### PR TITLE
Add redcore webservices decoupled code

### PR DESCRIPTION
**Pull request by Kixo**
webservices HAL + SOAP protocol all in one...

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# OSX
+.DS_Store
```
Comment (nibra) [inline .gitignore]: This kind of files belong to your own global .gitignore file (.DS_Store, Thumb.db, Desktop.ini, .idea/ and so on)

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# Github pages and Jekyll files
+/_site/
+/Gemfile
+/Gemfile.lock
+
+# phpunit
+tests/unit/phpunit.xml
+
+# composer
+composer.lock
```
Comment (nibra) [inline .gitignore]: As I learned at JaB, composer.lock has to be comitted.

File: .gitignore (hunk 1 above)
Comment (Kixo) [inline .gitignore]: ok

File: .gitignore (hunk 2 above)
Comment (Kixo) [inline .gitignore]: it is actually not needed, a leftover from redcore, so I will remove it

**Comment by fastslack** [comment]
@Kixo im trying to install it but seems that there is an issue with /api load:

**Pull request merged**